authors/apps/articles/views.py

```diff
@@ -84,4 +99,81 @@ def destroy(self, request, pk=None):
         article = get_object_or_404(queryset, pk=pk)
         article.delete()
         return Response({"message": "article deleted successfully"},
-                        status=status.HTTP_200_OK)
+                        status=status.HTTP_204_NO_CONTENT)
+
+
+class RatingAPIView(GenericAPIView):
+    queryset = Rating.objects.all()
+    serializer_class = RatingSerializer
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    # renderer_classes = (RatingJsonRenderer,)
+
+    def post(self, request, slug):
+        """POST request to rate an article."""
+        rating = request.data
+        article = get_article(slug)
+
+        if not article:
```

Comment:
This won't happen since in the absence of an article, an exception will be thrown (line 20)